engine/function/src/templates/Basic.ftl

```diff
@@ -1459,10 +1513,12 @@ public class Basic {
         final ArrayList<${pt.boxed}> orderedList = new ArrayList<>();
```

Comment:
We should audit the usages of `toArray()`. I think we need to be more cautious about our contracts re: copying.
1. L126: `replaceIfNullObj` as noted in a separate comment
2. L304: `arrayObj` might be returning the vector's own backing array. Is this OK? The JavaDoc doesn't specify. Do you need/want a version of `toArray` that guarantees it will always allocate a new result?
3. L530: `distinctObj` has the same kind of issue as `arrayObj` for single-element vectors.
4. L920: `forwardFillObj` as noted in a separate comment.
5. L1190: `array` has the same kind of issue as `arrayObj`
6. L1439: `distinct` has the same kind of issue as `distinctObj`
7. L1510: `ObjectVector<Boolean>` version of `distinct` has the same kind of issue as `distinctObj`